File: app/src/main/java/com/example/criminalintent/CrimeActivity.java
package com.example.criminalintent;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import java.util.UUID;

public class CrimeActivity extends AppCompatActivity {

    public static final String EXTRA_CRIME_ID = "com.example.criminalintent.crime_id";

    public static Intent newIntent(Context context, UUID crimeId) {
        Intent intent = new Intent(context, CrimeActivity.class);
        intent.putExtra(EXTRA_CRIME_ID, crimeId);
        return intent;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_crime);

        UUID crimeId = null;
        if (getIntent() != null) {
            Object extra = getIntent().getSerializableExtra(EXTRA_CRIME_ID);
            if (extra instanceof UUID) {
                crimeId = (UUID) extra;
            }
        }
        if (crimeId == null) return;

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.fragment_container, CrimeFragment.newInstance(crimeId))
                    .commit();
        }
    }
}

File: app/src/main/java/com/example/criminalintent/CrimeActivity.kt
package com.example.criminalintent

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import java.util.UUID

class CrimeActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_crime)

        val crimeId = intent.getSerializableExtra(EXTRA_CRIME_ID) as? UUID ?: return

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, CrimeFragment.newInstance(crimeId))
                .commit()
        }
    }

    companion object {
        const val EXTRA_CRIME_ID = "com.example.criminalintent.crime_id"

        fun newIntent(context: Context, crimeId: UUID): Intent {
            return Intent(context, CrimeActivity::class.java).apply {
                putExtra(EXTRA_CRIME_ID, crimeId)
            }
        }
    }
}
